packages/client/src/wrappers.tsx

```diff
@@ -241,8 +244,14 @@ export function DocsPage({
           <Text>This page could not be found.</Text>
         </HStack>
       ) : (
-        <Component {...pageProps} />
+        <Component {...pageProps} sourceFilePath={sourceFilePath} />
       )}
     </DocsContainer>
   );
 }
+
+function getMDXPathFromRoute(asPath: string, data: IRoutes): string {
+  // TODO: traverse data and find the correct `.mdx` path based on the asPath...
+
+  return 'foo.mdx';
+}
```

Comment:
This is kind of the missing peace. It also kind of feels weird that we have to traverse all the stuff for finding the source file based on the actual url. I still don't fully understand the route generation rules. The logic feels quite complex...